src/Storages/ObjectStorage/DataLakes/DataLakeStorageSettings.h

```diff
@@ -47,7 +47,7 @@ class SettingsChanges;
 
 #define DATA_LAKE_STORAGE_RELATED_SETTINGS(DECLARE, ALIAS) \
     DECLARE(Bool, allow_dynamic_metadata_for_data_lakes, true, R"(
```

Comment:
Let's move it here https://github.com/ClickHouse/ClickHouse/blob/daef0918845b18360b257269ac3329bbbd0178ab/src/Storages/ObjectStorage/DataLakes/DataLakeStorageSettings.h#L80